pkg/apis/infra/v1alpha1/cluster.go

```diff
@@ -52,6 +52,23 @@ type CustomClusterSpec struct {
 
 	// CNIConfig is the configuration for the CNI of the cluster.
 	CNI CNIConfig `json:"cni"`
+
+	// EnableVIP setting to "true" indicates the selection of kube-vip for achieving High Availability (HA).
+	EnableVIP bool `json:"enableVIP,omitempty"`
+
+	// VIPConfig is the config of VIP for HA when the EnableVIP is set to "true".
+	VIP VIPConfig `json:"vip,omitempty"`
```

Comment:
VIP is not clear, whose vip. Assume it is cluster, then what does cluster vip mean. We should consider carefully